node/blockAPI/baseBlock_test.go

```diff
@@ -0,0 +1,265 @@
+package blockAPI
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/node/mock"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/stretchr/testify/assert"
+)
+
+func createMockArgumentsWithTx(
+	srcShardId uint32,
+	destShardId uint32,
+	recvAddress string,
+	miniblockBytes []byte,
+	miniblockType block.Type,
+	txHash string,
+	txBytes []byte,
+	marshalizer *mock.MarshalizerFake,
+) baseAPIBockProcessor {
+	return baseAPIBockProcessor{
+		selfShardID: srcShardId,
+		marshalizer: marshalizer,
+		store: &mock.ChainStorerMock{
+			GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+				return &mock.StorerStub{
+					GetBulkFromEpochCalled: func(keys [][]byte, epoch uint32) (map[string][]byte, error) {
+						return map[string][]byte{txHash: txBytes}, nil
+					},
+					GetFromEpochCalled: func(key []byte, epoch uint32) ([]byte, error) {
+						return miniblockBytes, nil
+					},
+				}
+			},
+		},
+		historyRepo: &testscommon.HistoryRepositoryStub{
+			IsEnabledCalled: func() bool {
+				return false
+			},
+		},
+		unmarshalTx: func(txBytes []byte, txType transaction.TxType) (*transaction.ApiTransactionResult, error) {
+			var unmarshalledTx transaction.Transaction
+			_ = marshalizer.Unmarshal(&unmarshalledTx, txBytes)
+			return &transaction.ApiTransactionResult{
+				Tx:               &unmarshalledTx,
+				Type:             string(txType),
+				Nonce:            unmarshalledTx.Nonce,
+				Hash:             txHash,
+				MiniBlockType:    string(miniblockType),
+				SourceShard:      srcShardId,
+				DestinationShard: destShardId,
+				Receiver:         recvAddress,
+				Data:             []byte{},
+			}, nil
+		},
+	}
+}
+
+func TestBaseApiBlockProcessor_GetNormalTxFromMiniBlock(t *testing.T) {
+	t.Parallel()
+
+	epoch := uint32(1)
+	nonce := uint64(0)
+	sourceShardId := uint32(1)
```

Comment:
sourceShardID ----> use ID with the upper case instead of Id in all over the places

Reply:
done